modal/image.py

```diff
@@ -1299,6 +1299,68 @@ def build_dockerfile(version: ImageBuilderVersion) -> DockerfileSpec:
             gpu_config=parse_gpu_config(gpu),
         )
 
+    def uv_sync(
+        self,
+        lockfile: str = "./uv.lock",  # Path to lockfile
+        # Path to pyproject.toml. If not provided, then use file in the same directory as the lockfile
+        pyproject_toml: Optional[str] = None,
```

Comment:
j/w but how are you thinking about the relative importance of the project file vs the lockfile here?

Reply:
Actually having completely borked my local setup I've now learned about the "uv workspace" concept and I think that does provide a counterexample to the question of 

> Are there any uv workflows where the lockfile does not end up next to the project file?

since i gather that uv [expects](https://docs.astral.sh/uv/concepts/projects/workspaces/#workspace-layouts) the lockfile to be one directory up from the pyproject.toml in this setup.

FWIW this sort of thing is why I've been reticent to "support" uv in modal in ways that require modal to bake in uv's assumptions about project layout etc. as it feels like they're still experimenting with various ideas. Not sure exactly what the actionable implication is though.

Reply:
With uv workspaces with a root `pyproject.toml`:

```python
[project]
name = "my-app"
version = "0.1.0"
description = "An example of uv workspace"
readme = "README.md"
requires-python = ">=3.9"
dependencies = ["modal"]

[tool.uv.sources]
my-app = { workspace = true }

[tool.uv.workspace]
members = ["packages/*"]
```

It expects a file structure like this:

```bash
.
├── packages
│   ├── mod1
│   │   └── pyproject.toml
│   └── mod2
│       └── pyproject.toml
├── pyproject.toml
└── uv.lock
```

For us to properly support this, we'll have to parse the root `pyproject.toml` and understand to upload the other `pyproject.toml` and 

For the first iteration, I'll like to not support this and `[tool.uv.workspace]` members and error if it exist.